modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -128,5 +132,51 @@ object ConstructionService {
           }
         }
 
+    override def getPayloads(
+      operations: NonEmptyList[Operation],
+      maybeMetadata: Option[TransactionMetadata]
+    ): EitherT[F, ConstructionError, PayloadsResult] =
+      for {
+        _ <- EitherT.cond(operations.size == 2, (), IncorrectNumberOfOperations)
+
+        (positiveOperation, transactionAmount) <- EitherT.fromOption(
+          operations.collectFirst {
+            case op if op.amount.exists(!_.value.isNegative) =>
+              op.amount
+                .flatMap(_.value.toTransactionAmount.toOption)
+                .map((op, _))
+          }.flatten,
+          MissingPositiveAmountOperation
+        )
+
+        metadata <- EitherT.fromOption(maybeMetadata, MissingMetadata)
+
+        accountId <- EitherT.fromOption(positiveOperation.account, MissingAccountIdentifier)
+
+        dagTransaction =
+          DAGTransaction(
+            source = metadata.source,
+            destination = accountId.address,
+            amount = transactionAmount,
+            fee = metadata.fee,
+            parent = metadata.lastReference,
+            salt = metadata.salt.getOrElse(TransactionSalt(0L))
+          )
+
+        serializedTxn <-
+          KryoSerializer[F]
+            .serialize(dagTransaction)
+            .leftMap(_ => MalformedTransaction)
```

Comment:
imo `MalformedTransaction` is being returned when the user passed the malformed `hex` that should represent a transaction and cannot be deserialized. if the transaction cannot be serialized here, it's most likely a programatic error (because you create that transaction in a previous operation) and imo the error should be different